pkg/controller/certificates/certificate_request.go

```diff
@@ -531,51 +531,86 @@ func (c *certificateRequestManager) processCertificate(ctx context.Context, crt
 		return nil
 	}
 
-	// Check if the CertificateRequest is Ready, if it is not then we return
-	// and wait for informer updates to re-trigger processing.
-	if !apiutil.CertificateRequestHasCondition(existingReq, cmapi.CertificateRequestCondition{
-		Type:   cmapi.CertificateRequestConditionReady,
-		Status: cmapi.ConditionTrue,
-	}) {
-		log.Info("certificate request is not in a Ready state, waiting until CertificateRequest is issued")
-		// TODO: we need to handle failure states too once we have defined how we represent them
-		return nil
-	}
+	// Determine the status reason of the CertificateRequest and process accordingly
+	switch apiutil.CertificateRequestStatusReason(existingReq) {
+	// If the CertificateRequest exists but has failed then we check the failure
+	// time. If the failure time doesn't exist or is over an hour in the past
+	// then delete the request so it can be re-created on the next sync. If the
+	// failure time is less than an hour in the past then schedule this owning
+	// Certificate for a re-sync in an hour.
+	case cmapi.CertificateRequestReasonFailed:
+		if existingReq.Status.FailureTime == nil {
+			log.Info("the failed existing certificate request has no failure time so will be retried now")
+
+		} else if c.clock.Since(existingReq.Status.FailureTime.Time) > time.Hour {
```

Comment:
Sorry to be a pain, but this `if/else if/else` is killing my inner gopher.. not that important, but this *could* be written as:

```golang
case cmapi.CertificateRequestReasonFailed:
    if existingReq.Status.FailureTime == nil || c.clock.Since(existingReq.Status.FailureTime.Time) > time.Hour {
        return r.deleteCertificateRequest(existingReq)
    }
    // otherwise, schedule for re-processing in N minutes
```

Reply:
:heart: